axelor-base/src/main/java/com/axelor/apps/base/service/ABCAnalysisServiceImpl.java

```diff
@@ -156,37 +158,55 @@ protected String getProductFamilyQuery(){
         return "self.productFamily in (?1) AND self.productTypeSelect = ?2";
     }
 
+    protected void createAllABCAnalysisLine(ABCAnalysis abcAnalysis) throws AxelorException {
+        int offset = 0;
+
+        List<Product> productList;
+        Query<Product> productQuery = productRepository.all()
+                .filter("self.id IN (" + StringTool.getIdListString(getProductSet(abcAnalysis)) + ")");
+
+        while(!(productList = productQuery.fetch(FETCH_LIMIT, offset)).isEmpty()){
+            offset += productList.size();
+
```

Comment:
Find methods should be called here, because we need it when we call the JPA.clear method.

With this way we will call the find method only when is required instead of on each iteration in createABCAnalysisLine method.